flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamOverAggregate.scala

```diff
@@ -119,6 +152,60 @@ class DataStreamOverAggregate(
 
   }
 
+  def createTimeBoundedProcessingTimeOverWindow(inputDS: DataStream[Row]): DataStream[Row] = {
+
+    val overWindow: Group = logicWindow.groups.get(0)
+    val partitionKeys: Array[Int] = overWindow.keys.toArray
+    val namedAggregates: Seq[CalcitePair[AggregateCall, String]] = generateNamedAggregates
+
+    val index = overWindow.lowerBound.getOffset.asInstanceOf[RexInputRef].getIndex
+    val count = input.getRowType().getFieldCount()
+    val lowerboundIndex = index - count
+    
+    
+    val time_boundary = logicWindow.constants.get(lowerboundIndex).getValue2 match {
+      case _: java.math.BigDecimal => logicWindow.constants.get(lowerboundIndex)
+         .getValue2.asInstanceOf[java.math.BigDecimal].longValue()
+      case _ => throw new TableException("OVER Window boundaries must be numeric")
+    }
+
+    val (aggFields, aggregates) =  AggregateUtil.transformToAggregateFunctions(
+         namedAggregates.map(_.getKey),inputType, needRetraction = false)
+         
+         
+    // As we it is not possible to operate neither on sliding count neither
+    // on sliding time we need to manage the eviction of the events that
+    // expire ourselves based on the proctime (system time). 
+    
+    // get the output types
+    val rowTypeInfo = FlinkTypeFactory.toInternalRowTypeInfo(getRowType).asInstanceOf[RowTypeInfo]
+
+    val result: DataStream[Row] =
+      if (partitionKeys.nonEmpty) {
+        inputDS.keyBy(partitionKeys:_*)
+          .window(GlobalWindows.create())
+          .trigger(CountTrigger.of(1))
+          .evictor(TimeEvictor.of(Time.milliseconds(time_boundary))) 
```

Comment:
@fhueske Actually no - it works on the time that is set. Basically if the user operates on processing time than this is what it will use. It checks at the beginning of the eviction if the record has processing time and if so that it operates on it. I both verified it  manually as well as with tests.